src/routers/alpha-router/alpha-router.ts

```diff
@@ -1631,6 +1642,86 @@ export class AlphaRouter
     }
 
     if (!swapRouteRaw) {
+      // this is a well-intended scoped fix,
+      // because we already know that in all the reported cached routes invalid case,
+      // end user cannot get quote.
+      // therefore its safe to limit the cached routes cache invalidation when there;s no quote.
+      if (
+        Math.random() * 100 <
+        (this.cachedRoutesCacheInvalidationFixRolloutPercentage ?? 0)
+      ) {
+        // there's a possibility that cachedRoutes object is not null/undefined, but it could cause swapRouteFromCache to be null/undefined
+        // because such cached routes already becomes invalid.
+        // in that case then, we want to try swapRouteFromChain.
+        // If CachedRoutes is populated, then swapRouteFromChain is not being invoked at all in
+        // https://github.com/Uniswap/smart-order-router/blob/66927c4e4f7e75138df48afab1c33a6c6d8d8f15/src/routers/alpha-router/alpha-router.ts#L1469
+        if (
+          this.routeCachingProvider &&
+          routingConfig.writeToCachedRoutes &&
+          cacheMode !== CacheMode.Darkmode &&
+          // if cachedRoutes is defined, then swapRouteFromChain cannot be defined
+          // there's a possibility that swapRouteFromCache is undefined, in cached routes become invalid
+          // in that case, we want to make sure we swap from chain.
+          // we use optimisticCachedRoutes = false to make sure it's intent=caching path
+          cachedRoutes &&
```

Comment:
so this is the chance where the entry does exist, but the entry itself is null right? because `swapRouteFromCache` will by definition get created if `cachedRoutes` is true right?

Reply:
aw i see.. i don't see that's possible. becauses during the cache set in routing-api, individual route has to exist with the route id in https://github.com/Uniswap/routing-api/blob/main/lib/handlers/router-entities/route-caching/dynamo-route-caching-provider.ts#L388, we can sync more offline